vendor/k8s.io/kubernetes/cmd/kube-controller-manager/app/patch_version.go

```diff
@@ -0,0 +1,19 @@
+package app
+
+import (
+	"github.com/golang/glog"
+	apimachineryversion "k8s.io/apimachinery/pkg/version"
+)
+
+var (
+	// OpenshiftVersionInfo exposes openshift verison info to k8s packages
+	OpenshiftVersionInfo *apimachineryversion.Info
+)
+
+// GetOpenshiftVersion returns Overall Openshift codebase version
+func GetOpenshiftVersion(componentKey string) *apimachineryversion.Info {
+	if OpenshiftVersionInfo == nil {
+		glog.Fatalf("Can not find Openshift version for retrieving matching images. Openshift can be additionally started with %s environment variable to find matching image", componentKey)
```

Comment:
Why would we ever be setting this env var?  Why wouldn't the solution be "set the `--whatever` flag on kube-controller-manager" if they need something this doesn't do?

Reply:
Because then they will have to create pod definition of recycler themselves and put in on disk somewhere and make sure it is always available on controller-manager on restart etc.

I prefer that user just has to declare the environment variable for container image because in that case, we don't have to document what pod definition should look like etc.

Reply:
well - at least that is what I was thinking. documenting `--pv-recycler-pod-template-filepath-nfs` works too.

Reply:
Another reason was - those 2 flags only cover 2 cases of container image use, whereas using environment variable will cover all use cases(or any other container image that is referred in code).

Reply:
> or any other container image that is referred in code).

There should never be another one.  We don't want drift from upstream.



> Because then they will have to create pod definition of recycler themselves and put in on disk somewhere and make sure it is always available on controller-manager restart etc.

Is this something that changes often?  In 3.10, I'd expect it to be an ansible maintained file.

Reply:
> Is this something that changes often? In 3.10, I'd expect it to be an ansible maintained file.

The container image tag changes on each release, nothing else changes - afaik.I don't think the code of binary being called from container changes at all even if new versions are tagged. so one can workaround by tagging any random old version to "openshift/origin-recycler:v3.10" and it will still do the job.

Reply:
Also the container image name for OCP and origin are different (ose3/origin-recycler vs openshift/origin-recycler) and hence pod definition has to be different. I am not very sure about bringing this pod definition inside openshift-ansible, the feature is being deprecated in upstream and we can possibly workaround thiis whole versioning thing by always referring "latest".

Reply:
So just to clarify rather than using Environment variable you prefer we use the CLI args? Another thing we can do is - rather than logging fatal() error and exiting, we can simply set version to `latest` if openshift variable is not available. 

That should work for majority of cases and since container code does not changes between versions - the recycler remains same.